src/RouteCollector.php

```diff
@@ -27,13 +28,74 @@ public function __construct(RouteParser $routeParser, DataGenerator $dataGenerat
      * @param mixed  $handler
      */
     public function addRoute($httpMethod, $route, $handler) {
+        $route = $this->prependRouteWithRouteGroup($route);
         $routeDatas = $this->routeParser->parse($route);
         foreach ((array) $httpMethod as $method) {
             foreach ($routeDatas as $routeData) {
                 $this->dataGenerator->addRoute($method, $routeData, $handler);
             }
         }
     }
+
+    /**
+     * Prepends the route string with the route string specified in the route group.
+     *
+     * @param string $route
+     * @return string
+     */
+    protected function prependRouteWithRouteGroup($route)
+    {
+        return $this->prependRouteWithGroupsRoute($route, $this->currentRouteGroup);
+    }
+
+
+    /**
+     * Prepends the provided route with the route inside the provided route group.
+     *
+     * @param string $route
+     * @param null|\stdClass $group
+     * @return string
+     */
+    private function prependRouteWithGroupsRoute($route, $group)
+    {
+        if (is_object($group)) {
+            $route = "{$group->route}/" . ltrim($route, '/');
+        }
+        return $route;
+    }
+
+    /**
+     * Creates a new route group and returns it.
+     * If a previous group was passed it, it will prepend the groups route with the previous groups route.
+     *
+     * @param string $route
+     * @param callable $callback
+     * @param null|\stdClass $previousGroup
+     * @return \stdClass
+     */
+    private function createGroup($route, callable $callback, $previousGroup) {
+        $route = $this->prependRouteWithGroupsRoute($route, $previousGroup);
+        $group = new \stdClass();
```

Comment:
I don't think this structure is necessary, it should be sufficient to store only the group prefix in the route collector. The callback is only used once (in `addGroup`), so there is no need to store it.